frontend/app/api/auth/[...nextauth]/route.ts

```diff
@@ -71,14 +77,48 @@ export const authOptions: AuthOptions = {
         caller: 'callbacks/session',
         status: 200,
       })
-      return {
-        ...session,
-        user: {
-          id: token.sub,
-          name: user.name,
-          isAdmin: user.role === USER_ROLE.ADMIN,
-        },
+
+      if (user) {
+        return {
+          ...session,
+          user: {
+            id: token.sub,
+            name: user.name,
+            isAdmin: user.role === USER_ROLE.ADMIN,
+          },
+        }
+      } else {
+        return session
+      }
+    },
+    signIn: async ({ account, profile }) => {
+      if (account.provider === 'cognito') {
+        try {
+          const { sub: id, name, image } = profile
+          const existingUser = await getUser(id)
+
+          if (!existingUser) {
+            await fetch(
+              `${process.env.NEXT_PUBLIC_BACKEND_URL}/auth/google-signup`,
+              {
+                method: 'POST',
+                headers: {
+                  'Content-Type': 'application/json',
+                },
+                body: JSON.stringify({
+                  id,
+                  name,
+                  image,
+                }),
+              },
+            )
+          }
+          return true
+        } catch (err) {
+          throw new Error('エラーが発生しました')
```

Comment:
もう少しエラー内容を具体的にしたいですね。
他の部分でも同じようにエラーが発生しましただと
何のエラーなのかわからないので。

コードで全件検索してもわかりやすいように

```suggestion
          throw new Error('Error: siginIn error')
```

的な？

Reply:
修正しました！